src/serial.rs

```diff
@@ -319,21 +319,24 @@ macro_rules! hal {
                     // NOTE(unsafe) atomic read with no side effects
                     let isr = unsafe { (*$USARTX::ptr()).isr.read() };
 
+                    // NOTE(unsafe) write accessor for atomic writes with no side effects
+                    let icr = unsafe { &(*$USARTX::ptr()).icr };
+
                     Err(if isr.pe().bit_is_set() {
                         // Clear the bit by writing 1 to the clear register
-                        unsafe { (*$USARTX::ptr()).icr.write(|w| { w.pecf().set_bit() }) };
+                        icr.write(|w| { w.pecf().set_bit() });
```

Comment:
PECF offers a easy-to-understand [`.clear()` function](https://docs.rs/stm32f3xx-hal/0.4.2/stm32f3xx_hal/stm32/usart1/icr/struct.PECF_W.html)
This would (IMHO) make the comment redundant

Reply:
(This also holds for the other cases) below